docs/how_to_contribute/math/math_structure.md

```diff
@@ -0,0 +1,35 @@
+
+## Finite Fields
+
+[**Finite Fields**](https://mathworld.wolfram.com/FiniteField.html) inherit the Field class. There are two types of finite fields: [prime fields](https://mathworld.wolfram.com/PrimeField.html) and [extension fields](https://mathworld.wolfram.com/ExtensionField.html). A prime field is a finite field _GF(p)_ where _p_ is a prime number. An extension field, on the other hand, is a more complex structure that allows for the creation of larger fields from smaller ones through algebraic extensions.
+
+Finite fields currently implemented include [Goldilocks](https://xn--2-umb.com/22/goldilocks/) and [Mersenne-31](https://mathworld.wolfram.com/MersennePrime.html). More prime fields will be added in the future.
+
+Quadratic extension fields and cubic extension fields are types of extension fields, and have been used to implement `Fp2`, `Fp3`, `Fp4`, `Fp6`, and `Fp12`. These extended fields enable the extension of the base field for elliptic curves such as BLS12-381 and BN254.
+
+To add various finite fields, we use Bazel to automatically generate code. For instructions on how to add a new prime field, refer to this [doc](/docs/how_to_contribute/math/how_to_add_prime_fields.md).
+
+## Polynomials
+
+**Polynomials** inherit the Ring class. There are two types of polynomials: [univariate polynomials](https://mathworld.wolfram.com/UnivariatePolynomial.html) and [multivariate polynomials](https://mathworld.wolfram.com/MultivariatePolynomial.html).
+
+A univariate polynomial can be represented in coefficient form and evaluation form. A polynomial is further categorized into dense forms and sparse forms, which differ by the use of zero-coefficients. The dense forms are generally used, but the sparse forms are used in specific cases such as where only the max degree and the constant exist in a polynomial.
+
+Multivariate polynomials can be represented in multivariate sparse coefficient form and multilinear dense evaluation form. Since the multivariate dense coefficient form and multilinear spare evaluation form rarely exist, neither are implemented.
+
+## Elliptic Curves
+
+[**Elliptic Curves**](https://mathworld.wolfram.com/EllipticCurve.html) inherit the Group class. Elliptic curves come in various forms, but all currently implemented elliptic curves are in the Short Weierstrass form. Elliptic curves' points can be represented in four formats: Affine, Projective, Jacobian, and PointXYZZ.
+
+The currently implemented elliptic curves are BLS12-381, BN254, Secp256k1, and Pasta. Similar to finite fields, we generate code with Bazel's generator to add a wider variety of curves. For instructions on how to add a new elliptic curve, refer to this [doc](/docs/how_to_contribute/math/how_to_add_elliptic_curves.md).
+
+Since the result of AffinePoint + JacobianPoint is a JacobianPoint, the `ResultTy` of `AdditiveSemigroupTraits` for the elliptic curves is a JacobianPoint.
```

Comment:
I'm not too sure what this is trying to tell us-it seems to come out of nowhere

Reply:
We decided to add only the diagram in this PR and update this doc in another PR.